sycl/test-e2e/format.py

```diff
@@ -292,7 +292,15 @@ def get_extra_env(sycl_devices):
             conditions,
             recursion_limit=test.config.recursiveExpansionLimit,
         )
-        useExternalSh = False
+
+        # TODO: workaround for lit hanging when executing non-existent binary
```

Comment:
Another option is to only run pre-built binaries if that build task was 100% successful. Not sure if we want that, probably depends on the CI load patterns after the changes land and everything stabilizes.

Reply:
The reason i did not do this initially is because we can have a situation where the build fails on a test that would not be ran on a particular runner, so I think its still useful to run the testing.

Also for example if an unrelated change makes a single test on the build fail you would still want to see if your changes make any other tests fail at the run stage.

Reply:
That's one side of a coin. Another is that in 99% cases the author will upload an updated patch fixing those e2e-build issues and it's not unlikely that e2e-run-only results will be "wasted". If we can ensure that e2e-build failure is reported fast enough, then not running e2e in this scenario might improve the overall CI responsiveness.

Reply:
imo i like the CI still running as much as possible even if something fails

Reply:
I think another thing to consider in the scenario I highlighted is what we currently do when that occurs. Usually we'll point to a ticket and comment that the failure is unrelated and ask gatekeepers to merge as everything else passes. However if we don't run the testing we won't have this information, and effectively block the progress in such PRs.

Reply:
I think the "flaky" fails are happening 99.9% of the time on execution and not when building, so it might not be that bad. We'd just have to be quicker with the reverts when something does break. That said, if @sarnex wants to run as much as possible, then do that.